manim/utils/rate_functions.py

```diff
@@ -115,3 +119,180 @@ def exponential_decay(t, half_life=0.1):
     # The half-life should be rather small to minimize
     # the cut-off error at the end
     return 1 - np.exp(-t / half_life)
+
+
+def ease_in_sin(t):
+    return 1 - np.cos((t * np.pi) / 2)
+
+
+def ease_out_sin(t):
+    return np.sin((t * np.pi) / 2)
+
+
+def ease_in_out_sin(t):
+    return -(np.cos(np.pi * t) - 1) / 2
+
+
+def ease_in_quad(t):
+    return t * t
+
+
+def ease_out_quad(t):
+    return 1 - (1 - t) * (1 - t)
+
+
+def ease_in_out_quad(t):
+    return 2 * t * t if t < 0.5 else 1 - pow(-2 * t + 2, 2) / 2
+
+
+def ease_in_cubic(t):
+    return t * t * t
+
+
+def ease_out_cubic(t):
+    return 1 - pow(1 - t, 3)
+
+
+def ease_in_out_cubic(t):
+    return 4 * t * t * t if t < 0.5 else 1 - pow(-2 * t + 2, 3) / 2
+
+
+def ease_in_quart(t):
+    return t * t * t * t
+
+
+def ease_out_quart(t):
+    return 1 - pow(1 - t, 4)
+
+
+def ease_in_out_quart(t):
+    return 8 * t * t * t * t if t < 0.5 else 1 - pow(-2 * t + 2, 4) / 2
+
+
+def ease_in_quint(t):
+    return t * t * t * t * t
+
+
+def ease_out_quint(t):
+    return 1 - pow(1 - t, 5)
+
+
+def ease_in_out_quint(t):
+    return 16 * t * t * t * t * t if t < 0.5 else 1 - pow(-2 * t + 2, 5) / 2
```

Comment:
IMO the 5 `t`s look very unreadable but I thought to keep it as it is in that reference site, in case a dev in the future might have to refer back to the site (www.easings.net).

Reply:
I think it's fine.